components/script/dom/texttrack.rs

```diff
@@ -0,0 +1,139 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use crate::dom::bindings::codegen::Bindings::TextTrackBinding::{
+    self, TextTrackKind, TextTrackMethods, TextTrackMode,
+};
+use crate::dom::bindings::error::{Error, ErrorResult};
+use crate::dom::bindings::reflector::{reflect_dom_object, DomObject};
+use crate::dom::bindings::root::{DomRoot, MutNullableDom};
+use crate::dom::bindings::str::DOMString;
+use crate::dom::eventtarget::EventTarget;
+use crate::dom::texttrackcue::TextTrackCue;
+use crate::dom::texttrackcuelist::TextTrackCueList;
+use crate::dom::window::Window;
+use dom_struct::dom_struct;
+use std::cell::Cell;
+
+#[dom_struct]
+pub struct TextTrack {
+    eventtarget: EventTarget,
+    kind: TextTrackKind,
+    label: String,
+    language: String,
+    id: String,
+    mode: Cell<TextTrackMode>,
+    cue_list: MutNullableDom<TextTrackCueList>,
+}
+
+impl TextTrack {
+    pub fn new_inherited(
+        kind: TextTrackKind,
+        label: DOMString,
+        language: DOMString,
+        mode: TextTrackMode,
+    ) -> TextTrack {
+        TextTrack {
+            eventtarget: EventTarget::new_inherited(),
+            kind: kind,
+            label: label.into(),
+            language: language.into(),
+            id: "".to_owned(),
```

Comment:
I wasn't sure what to do here. The spec states.

> the track's identifier, if it has one, or the empty string otherwise

But in gecko the TextTrack class does not have an `id` member. For now I've added `id` to `new` and `new_inherited`